readthedocs/proxito/views.py

```diff
@@ -209,22 +187,70 @@ def serve_docs(
             current_project.translations.all(), language=lang_slug
         )
 
+    # Handle single version by grabbing the default version
+    if final_project.single_version:
+        version_slug = final_project.get_default_version()
+
     # ``final_project`` is now the actual project we want to serve docs on,
     # accounting for:
     # * Project
     # * Subproject
     # * Translations
 
+    return final_project, lang_slug, version_slug, filename
+
+
+def serve_docs(
+        request,
+        project_slug=None,
+        subproject_slug=None,
+        lang_slug=None,
+        version_slug=None,
+        filename='',
+):
+    """Take the incoming parsed URL's and figure out what file to serve."""
+
+    log.debug(
+        'project=%s, subproject=%s, lang_slug=%s, version_slug=%s, filename=%s',
+        project_slug, subproject_slug, lang_slug, version_slug, filename
+    )
```

Comment:
I'd move this after `_get_project_data_from_request` and print the final ones (or both).